unity/Assets/Maroon/reusablePrefabs/NewRuler/Scripts/RulerLogic.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Maroon.Physics;
+using Maroon.Physics.CoordinateSystem;
+using UnityEngine;
+using Maroon.GlobalEntities;
+
+namespace Assets.Maroon.reusablePrefabs.NewRuler.Scripts
+{
+    public class RulerLogic : MonoBehaviour, IResetObject
+    {
+        public GameObject RulerStart;
```

Comment:
use the type Pin (old RulerNotify) so that you don't need to call the GetComponent method